web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -233,14 +233,16 @@ async function setUpPtyProcess(
   if (doc.kind === 'doc.terminal_tsh_node') {
     ctx.usageService.captureProtocolUse(clusterUri, 'ssh', doc.origin);
   }
-  if (doc.kind === 'doc.terminal_tsh_kube') {
+  if (doc.kind === 'doc.terminal_tsh_kube' || doc.kind === 'doc.gateway_kube') {
     ctx.usageService.captureProtocolUse(clusterUri, 'kube', doc.origin);
   }
 
   const openContextMenu = () => ctx.mainProcessClient.openTerminalContextMenu();
 
   const refreshTitle = async () => {
-    if (cmd.kind !== 'pty.shell') {
+    // TODO(ravicious): Enable updating cwd in doc.gateway_kube titles by
+    // moving title-updating logic to DocumentsService.
```

Comment:
I know I suggested that in one of the previous PRs, but let's expand the comment since I probably won't remember what I meant in six months. :D

> The logic behind updating the title should be encapsulated in a single place, so that useDocumentTerminal doesn't need to know the details behind the title of each document kind.